modules/docs/mdx/12.0-UPGRADE-GUIDE.mdx

```diff
@@ -188,6 +190,14 @@ from Main instead.
 
 ---
 
+## TypeScript
+
+We've upgrade to TypeScript 5.0 and make use of
+[const Type Parameters](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#const-type-parameters).
+You will need to upgrade to TypeScript 5.0+ to avoid TypeScript syntax errors. TypeScript does not
+follow semver, so 5.0 doesn't mean a large breaking change from 4.9. TypeScript doesn't have a
+`x.10` release, they bump the `x.9` to `{x+1}.0`.
```

Comment:
```suggestion
We've upgraded to TypeScript 5.0 to make use of
[const Type Parameters](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#const-type-parameters).
You will need to upgrade to TypeScript 5.0+ to avoid any TypeScript syntax errors. TypeScript does not
follow semver, so 5.0 doesn't mean a large breaking change from 4.9. TypeScript doesn't have a
`x.10` release, they bump the `x.9` to `{x+1}.0`.
```